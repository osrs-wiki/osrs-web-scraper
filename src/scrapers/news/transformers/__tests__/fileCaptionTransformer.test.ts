import {
  MediaWikiBreak,
  MediaWikiBuilder,
  MediaWikiContent,
  MediaWikiFile,
  MediaWikiLink,
  MediaWikiText,
  MediaWikiTOC,
} from "@osrs-wiki/mediawiki-builder";

import NewsFileCaptionTransformer from "../fileCaptionTransformer";

describe("NewsFileCaptionTransformer", () => {
  it("should combine the adjacent MediaWikiFile, MediaWikiBreak and MediaWikiImage", () => {
    const originalContent: MediaWikiContent[] = [
      new MediaWikiFile("image"),
      new MediaWikiBreak(),
      new MediaWikiText([new MediaWikiText("caption", { italics: true })]),
    ];
    const transformed = new NewsFileCaptionTransformer().transform(
      originalContent
    );
    expect(
      new MediaWikiBuilder().addContents(transformed).build()
    ).toMatchSnapshot();
  });

  it("should combine the adjacent MediaWikiFile, MediaWikiBreak and MediaWikiImage with surrounding content", () => {
    const originalContent: MediaWikiContent[] = [
      new MediaWikiTOC(),
      new MediaWikiBreak(),
      new MediaWikiLink("test", "test"),
      new MediaWikiBreak(),
      new MediaWikiFile("image"),
      new MediaWikiBreak(),
      new MediaWikiText([new MediaWikiText("caption", { italics: true })]),
      new MediaWikiBreak(),
      new MediaWikiText("You can also discuss this update on our"),
      new MediaWikiBreak(),
    ];
    const transformed = new NewsFileCaptionTransformer().transform(
      originalContent
    );
    expect(
      new MediaWikiBuilder().addContents(transformed).build()
    ).toMatchSnapshot();
  });

  it("should combine the adjacent MediaWikiFile, MediaWikiBreak and MediaWikiImage with surrounding italics content", () => {
    const originalContent: MediaWikiContent[] = [
      new MediaWikiTOC(),
      new MediaWikiBreak(),
      new MediaWikiLink("test", "test"),
      new MediaWikiBreak(),
      new MediaWikiFile("image"),
      new MediaWikiBreak(),
      new MediaWikiText(
        [
          new MediaWikiText("caption"),
          new MediaWikiLink("linkText", "linkTarget"),
          new MediaWikiBreak(),
        ],
        {
          italics: true,
        }
      ),
      new MediaWikiBreak(),
      new MediaWikiText("You can also discuss this update on our", {
        italics: true,
      }),
      new MediaWikiBreak(),
    ];
    const transformed = new NewsFileCaptionTransformer().transform(
      originalContent
    );
    expect(
      new MediaWikiBuilder().addContents(transformed).build()
    ).toMatchSnapshot();
  });
});
