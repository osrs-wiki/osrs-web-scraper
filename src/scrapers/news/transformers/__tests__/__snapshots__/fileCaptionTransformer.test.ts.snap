// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsFileCaptionTransformer should combine the adjacent MediaWikiFile, MediaWikiBreak and MediaWikiImage 1`] = `
"[[File:image|thumb|''caption'']]
"
`;

exports[`NewsFileCaptionTransformer should combine the adjacent MediaWikiFile, MediaWikiBreak and MediaWikiImage with surrounding content 1`] = `
"__TOC__

[[test|test]]
[[File:image|thumb|''caption'']]

You can also discuss this update on our
"
`;

exports[`NewsFileCaptionTransformer should not combine non-adjacent MediaWikiFile, MediaWikiBreak and MediaWikiText 1`] = `
"__TOC__

[[File:image]]
test

''testitalics''
''The Old School Team.''"
`;
